MSAL/src/native_auth/controllers/MSALNativeAuthBaseController.swift

```diff
@@ -28,11 +28,14 @@ class MSALNativeAuthBaseController {
 
     typealias TelemetryInfo = (event: MSIDTelemetryAPIEvent?, context: MSALNativeAuthRequestContext)
     let clientId: String
+    weak var application: MSALNativeAuthPublicClientApplication?
```

Comment:
why is it needed for all the controllers to hold an instance of `MSALNativeAuthPublicClientApplication`?

Reply:
Not anymore. Please take a look on the new PR; https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/2159